staging/src/k8s.io/apiserver/pkg/server/genericapiserver_graceful_termination_test.go

```diff
@@ -198,15 +220,27 @@ func TestGracefulTerminationWithKeepListeningDuringGracefulTerminationDisabled(t
 		t.Fatal("Expected the AfterShutdownDelayDuration verification step to execute")
 	}
 
-	// step 4: wait for the HTTP Server listener to have stopped
+	// step 4: wait for the AfterShutdownDelayDuration event
+	select {
+	case <-s.lifecycleSignals.AfterShutdownDelayDuration.Signaled():
+	case <-time.After(5 * time.Second):
```

Comment:
nit: let's switch 5s to `wait.ForeverTestTimeout`

[I see 5s is used elsewhere too so I suggesst switch elsewhere too]